### PR TITLE
Add javadoc for the recording fakes.

### PR DESCRIPTION
**Pull request by wiarlawd**
Other changes:
* Improve links in related interfaces and classes.
* Fix javadoc errors and warnings.
* Change AbstractDocIdPusher javadoc to @inheritDoc for context.

### CODE REVIEW

File: src/com/google/enterprise/adaptor/testing/RecordingDocIdPusher.java [hunk 1]
```diff
@@ -35,7 +36,7 @@
 
 /**
  * A fake implementation of {@link DocIdPusher} that simply records
- * the values it receives.
+ * the values it receives. This implementation is not thread-safe.
```
Comment (PapaJoltOmega) [inline src/com/google/enterprise/adaptor/testing/RecordingDocIdPusher.java:39]: Much appreciated.

**Review by PapaJoltOmega — APPROVED** [review]
Thank you

File: src/com/google/enterprise/adaptor/CommandStreamParser.java [hunk 2]
```diff
@@ -136,7 +136,7 @@
  * immediately preceding metadata-name<p>
  *
  * "param-name=" -- specifies a parameter key, to be followed by a parameter-value.
- * Parameters are supplied to {@link MetadataTransforms} for use when making
+ * Parameters are supplied to {@link MetadataTransform MetadataTransforms} for use when making
```
Comment (BrettMichaelJohnson) [inline src/com/google/enterprise/adaptor/CommandStreamParser.java:139]: 80 cols?

Comment (wiarlawd) [reply src/com/google/enterprise/adaptor/CommandStreamParser.java:139]: This file and few others from the command line adaptors are 100 cols.

**Review by BrettMichaelJohnson — APPROVED** [review]

**Pull request merged**